Content.Client/Administration/UI/CustomControls/CommandButton.cs

```diff
@@ -4,52 +4,70 @@
 using Robust.Client.UserInterface;
 using Robust.Client.UserInterface.Controls;
 
-namespace Content.Client.Administration.UI.CustomControls
+namespace Content.Client.Administration.UI.CustomControls;
+
+[Virtual]
+public class CommandButton : Button, IDocumentTag
 {
-    [Virtual]
-    public class CommandButton : Button, IDocumentTag
+    private ConfirmWindow? _window;
+
+    public string? Command { get; set; }
+    public string? RequiresConfirm { get; set; }
+
+
+    public CommandButton()
     {
-        public string? Command { get; set; }
+        OnPressed += Execute;
+    }
 
-        public CommandButton()
-        {
-            OnPressed += Execute;
-        }
+    protected virtual bool CanPress()
+    {
+        return string.IsNullOrEmpty(Command) ||
+            IoCManager.Resolve<IClientConGroupController>().CanCommand(Command.Split(' ')[0]);
+    }
 
-        protected virtual bool CanPress()
+    protected override void EnteredTree()
+    {
+        if (!CanPress())
         {
-            return string.IsNullOrEmpty(Command) ||
-                   IoCManager.Resolve<IClientConGroupController>().CanCommand(Command.Split(' ')[0]);
+            Visible = false;
         }
+    }
 
-        protected override void EnteredTree()
-        {
-            if (!CanPress())
-            {
-                Visible = false;
-            }
-        }
+    private void Confirm()
+    {
+        if (string.IsNullOrEmpty(Command))
+            return;
 
-        protected virtual void Execute(ButtonEventArgs obj)
-        {
-            // Default is to execute command
-            if (!string.IsNullOrEmpty(Command))
-                IoCManager.Resolve<IClientConsoleHost>().ExecuteCommand(Command);
+        _window = new ConfirmWindow(Command);
+        _window.OpenCentered();
+    }
+
+    protected virtual void Execute(ButtonEventArgs obj)
+    {
+        if (string.IsNullOrEmpty(Command))
+            return;
+
+        if (RequiresConfirm != null) {
+            Confirm();
+            return;
         }
 
-        public bool TryParseTag(Dictionary<string, string> args, [NotNullWhen(true)] out Control? control)
+        IoCManager.Resolve<IClientConsoleHost>().ExecuteCommand(Command);
+    }
+
+    public bool TryParseTag(Dictionary<string, string> args, [NotNullWhen(true)] out Control? control)
+    {
+        if (args.Count != 2 || !args.TryGetValue("Text", out var text) || !args.TryGetValue("Command", out var command))
```

Comment:
You've changed the number of arguments here - args.Count is allowed be 3. Does this code even execute?